src/main/java/io/xdb/core/persistence/Persistence.java

```diff
@@ -1,3 +1,98 @@
 package io.xdb.core.persistence;
 
-public class Persistence {}
+import com.google.common.collect.ImmutableList;
+import io.xdb.core.exception.GlobalAttributeNotFoundException;
+import io.xdb.gen.models.GlobalAttributeTableRowUpdate;
+import io.xdb.gen.models.LoadGlobalAttributeResponse;
+import io.xdb.gen.models.TableColumnValue;
+import io.xdb.gen.models.TableReadResponse;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Persistence {
+
+    // table: { columnKey: columnValue }
+    private final Map<String, Map<String, String>> globalAttributes = new HashMap<>();
+    // table: { columnKey: columnValue }
+    private final Map<String, Map<String, String>> globalAttributesToUpdate = new HashMap<>();
+
+    public Persistence(final LoadGlobalAttributeResponse loadGlobalAttributeResponse) {
+        if (loadGlobalAttributeResponse == null) {
+            return;
+        }
+
+        final List<TableReadResponse> tableResponses = loadGlobalAttributeResponse.getTableResponses() == null
+            ? ImmutableList.of()
+            : loadGlobalAttributeResponse.getTableResponses();
+        for (final TableReadResponse tableResponse : tableResponses) {
+            if (!globalAttributes.containsKey(tableResponse.getTableName())) {
+                globalAttributes.put(tableResponse.getTableName(), new HashMap<>());
+            }
+
+            final List<TableColumnValue> columns = tableResponse.getColumns() == null
+                ? ImmutableList.of()
+                : tableResponse.getColumns();
+            for (final TableColumnValue column : columns) {
+                globalAttributes.get(tableResponse.getTableName()).put(column.getDbColumn(), column.getDbQueryValue());
+            }
+        }
+    }
+
+    /**
+     * Get a global attribute.
+     *
+     * @param table
+     * @param columnKey
+     * @return  column value.
+     */
+    public String getGlobalAttribute(final String table, final String columnKey) {
+        if (!globalAttributes.containsKey(table)) {
+            throw new GlobalAttributeNotFoundException(
+                String.format("Table %s does not exist within the global attributes", table)
+            );
+        }
+        if (!globalAttributes.get(table).containsKey(columnKey)) {
+            throw new GlobalAttributeNotFoundException(
+                String.format(
+                    "Column key %s does not exist in the table %s within the global attributes",
+                    columnKey,
+                    table
+                )
+            );
+        }
+        return globalAttributes.get(table).get(columnKey);
+    }
+
+    /**
+     * Upsert a global attribute. If the table and column key exists, update the value. Else, insert a new global attribute.
```

Comment:
From the description here, I realize maybe we can remove the concept of global attribute, and use the table and table row directly. Because users have to specify table name, columns anyway, mixing the concepts global attributes and table definition together might cause some confustion. WDYT?

Reply:
There are a lot of thoughts around that. let's talk in zoom. 
A short answer is that user will be much worse experience of using that. For a column to insert, there is value, column name and also a type hint to convert the value into a DB friendly value to insert. Similar to return and convert the value back to read. 
For a schema, there is a need to define the primary key name in the schema. And then client need to provide the provide primary key value for each table on starting the process. 

I would love to simplify, and tried very very hard on that already.